src/react-image-lightbox.js

```diff
@@ -238,15 +244,10 @@ class ReactImageLightbox extends Component {
       this.moveRequested = false;
 
       // Load any new images
-      this.loadAllImages(nextProps);
+      this.loadAllImages(this.props);
     }
   }
 
-  shouldComponentUpdate() {
```

Comment:
I think this should be put back

Reply:
@wuweiweiwu youp. But it just got problems with updating - I can't figure out what's wrong. 
** componentDidUpdate** is not equal to ** componentWillReceiveProps**, so because of this ** shouldComponentUpdate** doesn't work as expected. I need to time to think about. What do you think about it?

Reply:
@mmarkelov thats a difficult problem. I think a better comparison is `getDerivedStateFromProps` is a replacement for `componentWillRecieveProps` 

There are some guides here https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
@wuweiweiwu yeah! But in **getDerivedStateFromProps** we cannot access to **this**